tests/test_datfromwav.py

```diff
@@ -0,0 +1,61 @@
+import bark, pytest, numpy as np, os.path
+from scipy.io import wavfile
+from bark.io import datfromwav as dfw
+
+def test_dat_from_wav_int(tmpdir):
+	
+#1.  Create wav file
+	# dtype: int array
+
+	fname_wav = os.path.join(tmpdir.strpath, 'test.wav')
+	arr_int = np.random.randint(1000, size=(1,100))
+	rate = 48000
+	wavfile.write(fname_wav, rate, arr_int)
+	assert os.path.exists(fname_wav), 'File Not Found: test.wav '
+
+
+	
+#2. Generate .dat and .dat.meta.yaml file
+	#Give .dat file path
+	fname_dat = os.path.join(tmpdir.strpath, 'test.dat')
+	attrs = {"name": "hello bark", "project": "bark"}
+	dat_file = dfw.dat_from_wav(fname_wav, fname_dat, **attrs)
+	assert os.path.exists(fname_dat), 'File Not Found: test.dat'
+	assert os.path.exists(os.path.join(tmpdir.strpath, 'test.dat.meta.yaml')), 'File Not Found: test.dat.meta.yaml'
+
+#3. Compare data, rate in .dat file
+	assert np.array_equal(arr_int, bark.read_sampled(fname_dat).data), 'Data in .wav and .dat files does not match' 
+	assert rate == bark.read_sampled(fname_dat).sampling_rate, 'Sampling rates does not match'
+	assert 'hello bark' == bark.read_sampled(fname_dat).attrs['name'], 'name attribute does not match'
+	assert 'bark' == bark.read_sampled(fname_dat).attrs['project'], 'project attribute does not match'
+	
+
+	
+
+def test_dat_from_wav_float(tmpdir):
```

Comment:
Large proportions of these two test functions are identical. Can you leverage that to avoid repeating yourself?

Reply:
Thanks for the good read @kylerbrown . It says a lot about latest test practices in python. However b/c this is a small independent test, it appears to me that by having a helper function to do the repetitive tasks for these 2 test functions might be a suitable fix for leveraging the replicative nature of these test scenarios. "I may be falling short in my understanding about fixtures." If this general approach of having a helper function rather than going for fixture decorator doesn't seem suitable then please let me know.
@gfetterman Since these 2 test cases differ only in dtype of numpy arrays, I decided to segregate this nature in two different functions suffixed by int/float(numpy dtypes) so that the user can comprehand which dtype failed.

Reply:
P.S. I could not think of a better name other than "helper_test". My bad!